### ISSUE TITLE
Hand Mapping not good

### ISSUE THREAD

**Issue by MuhammadAasharibNawshad** [AmyangXYZ/MiKaPo#2]
Hi, thankyou for the excellent repo. i was checking out performance of this repo. I have found that hand and fingers mapping is not good. Can you please tell how to improve hand and fingers mapping?

**Comment by AmyangXYZ** [comment]
If you see the white debug scene match your pose, then the problem is from my finger rotation calculation function. Hand is a little more complicated than main body calculation, but I will try to improve it. Will let you know when I fix this issue.

**Comment by MuhammadAasharibNawshad** [comment]
Ok. Just for your references here are my observations for the cases of performance deterioration and need improvement.
1) when i flip my hand from frontside to backside or vice versa.
2) when i bend/fold fingers to make fist
3) when hand goes near neck/ head the hand start moving arbitary.
4) when i show side of hand, it is not reflected in avatar

**Comment by liudongpu** [comment]
hi ,fix the issue?
''' 
      // setBoneRotation(getBone("左親指１"), result.left_thumb_cmc)
      // setBoneRotation(getBone("左親指２"), result.left_thumb_mcp)
      // setBoneRotation(getBone("左人指１"), result.left_index_finger_mcp)
      // setBoneRotation(getBone("左人指２"), result.left_index_finger_pip)
      // setBoneRotation(getBone("左人指３"), result.left_index_finger_dip)
      // setBoneRotation(getBone("左中指１"), result.left_middle_finger_mcp)
      // setBoneRotation(getBone("左中指２"), result.left_middle_finger_pip)
      // setBoneRotation(getBone("左中指３"), result.left_middle_finger_dip)
      // setBoneRotation(getBone("左薬指１"), result.left_ring_finger_mcp)
      // setBoneRotation(getBone("左薬指２"), result.left_ring_finger_pip)
      // setBoneRotation(getBone("左薬指３"), result.left_ring_finger_dip)
      // setBoneRotation(getBone("左小指１"), result.left_pinky_finger_mcp)
      // setBoneRotation(getBone("左小指２"), result.left_pinky_finger_pip)
      // setBoneRotation(getBone("左小指３"), result.left_pinky_finger_dip)

      // setBoneRotation(getBone("右親指１"), result.right_thumb_cmc)
      // setBoneRotation(getBone("右親指２"), result.right_thumb_mcp)
      // setBoneRotation(getBone("右人指１"), result.right_index_finger_mcp)
      // setBoneRotation(getBone("右人指２"), result.right_index_finger_pip)
      // setBoneRotation(getBone("右人指３"), result.right_index_finger_dip)
      // setBoneRotation(getBone("右中指１"), result.right_middle_finger_mcp)
      // setBoneRotation(getBone("右中指２"), result.right_middle_finger_pip)
      // setBoneRotation(getBone("右中指３"), result.right_middle_finger_dip)
      // setBoneRotation(getBone("右薬指１"), result.right_ring_finger_mcp)
      // setBoneRotation(getBone("右薬指２"), result.right_ring_finger_pip)
      // setBoneRotation(getBone("右薬指３"), result.right_ring_finger_dip)
      // setBoneRotation(getBone("右小指１"), result.right_pinky_finger_mcp)
      // setBoneRotation(getBone("右小指２"), result.right_pinky_finger_pip)
      // setBoneRotation(getBone("右小指３"), result.right_pinky_finger_dip)
'''

Or if there are relevant solutions, I will try to solve this problem.     
thanks

**Comment by MuhammadAasharibNawshad** [comment]
yes please! if you are able to solve the issue, then please feel free to contribute

**Comment by AmyangXYZ** [comment]
I figured out the solutions, all weird rotations will be fixed this weekend.
